### PR TITLE
Lawrence Whalen

### PR DESCRIPTION
**Pull request by LawrenceWhalen**

### CODE REVIEW

**Comment by jamogriff** [comment]
Quite the commit message!

**Comment by LawrenceWhalen** [comment]
I panicked when I hit the submit button early and deleted that branch. Then I created a new clone and managed to copy in 1.5 times as much code without noticing. I may have been slightly panicked.

File: lib/venue.rb [hunk 1]
```diff
@@ -1,2 +1,27 @@
...
+    @capacity = capacity
+    @patrons  = []
+  end
+
+  def add_patron(patron_name)
+    @patrons.push(patron_name)
+  end
+
+  def yell_at_patrons
+    @patrons.map { |patron| patron.upcase!  }
```
Comment (memcmahon) [inline lib/venue.rb:17]: I would avoid using the destructive `!` because it changes the @patrons array itself.

File: lib/venue.rb [hunk 2]
```diff
@@ -1,2 +1,27 @@
...
+  end
+
+  def over_capacity
+    @patrons.count > @capacity
+  end
+
+  def kick_out
+    @patrons.shift(@patrons.count - @capacity)
```
Comment (memcmahon) [inline lib/venue.rb:25]: Nice use of shift with an argument here!

File: spec/venue_spec.rb [hunk 3]
```diff
@@ -44,12 +39,47 @@
...
+
+  describe '#kick_out' do
+    it 'can kick out patrons until the venue is not over capacity' do
+      venue = Venue.new('Samson', 3)
+      venue.add_patron('Zack')
+      venue.add_patron('Zach')
+      venue.add_patron('Zip')
+      venue.add_patron('Zane')
+      venue.add_patron('Zales')
+      venue.add_patron('Tim')
```
Comment (memcmahon) [inline spec/venue_spec.rb:78]: 😆  'Tim'

File: lib/venue.rb [hunk 4]
```diff
@@ -1,2 +1,27 @@
...
+
+  def add_patron(patron_name)
+    @patrons.push(patron_name)
+  end
+
+  def yell_at_patrons
+    @patrons.map { |patron| patron.upcase!  }
+  end
+
+  def over_capacity
```
Comment (memcmahon) [inline lib/venue.rb:20]: Keep a close eye on the interaction pattern, we wanted this method to be called `over_capacity?`

**Review by memcmahon — COMMENTED** [review]
Great job!  Take a look at my notes and let me know if you have any questions.

Score: 4

**Pull request closed without merging**